### PR TITLE
Improve harness for conformance tests

### PR DESCRIPTION
**Pull request by anvacaru**
Updating the handler for conformance tests to support skipping individual tests, as suggested in https://github.com/runtimeverification/evm-semantics/pull/2491#discussion_r1646476742.

A dict is now created in the `_skipped_tests` method, where for each test file we store the skipped tests.
The `_test` method is also updated to skip a test if it is present in the generated dict.

Changes:
    The `failing.llvm` and `slow.llvm` lists are now encoded as `file_name,test_name`.

### CODE REVIEW

File: kevm-pyk/src/tests/integration/test_conformance.py [hunk 1]
```diff
@@ -37,7 +34,8 @@ def _test(gst_file: Path, schedule: str, mode: str, chainid: int, usegas: bool)
         gst_data = json.load(f)
 
     for test_name, test in gst_data.items():
-        _LOGGER.info(f'Running test: {gst_file} - {test_name}')
```
Comment (tothtamas28) [inline kevm-pyk/src/tests/integration/test_conformance.py:40]: I guess we still need this for debugging which test failed in a given GST file.

File: kevm-pyk/src/tests/integration/test_conformance.py [hunk 2]
```diff
@@ -57,13 +55,20 @@ def _assert_exit_code_zero(pattern: Pattern) -> None:
...
-def _skipped_tests() -> set[Path]:
-    def read_test_list(path: Path) -> list[Path]:
-        return [REPO_ROOT / test_path for test_path in path.read_text().splitlines()]
+def _skipped_tests() -> dict[Path, list[str]]:
+    slow_tests = read_csv_file(REPO_ROOT / 'tests/slow.llvm')
+    failing_tests = read_csv_file(REPO_ROOT / 'tests/failing.llvm')
+    skipped: dict[Path, list[str]] = defaultdict(list)
+    for test_file, test in slow_tests + failing_tests:
+        test_file = TEST_DIR / test_file
+        skipped[test_file].append(test)
```
Comment (tothtamas28) [inline kevm-pyk/src/tests/integration/test_conformance.py:64]: You can eliminate `defaultdict` using `dict.setdefault`:

```suggestion
    skipped: dict[Path, list[str]] = {}
    for test_file, test in slow_tests + failing_tests:
        test_file = TEST_DIR / test_file
        skipped.setdefault(test_file, []).append(test)
```

File: tests/slow.llvm [hunk 3]
```diff
@@ -1,2 +1,90 @@
...
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d8g0v0_Berlin
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d8g0v0_Istanbul
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d8g0v0_London
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d8g0v0_Merge
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d8g0v0_Shanghai
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d9g0v0_Berlin
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d9g0v0_Istanbul
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d9g0v0_London
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d9g0v0_Merge
+BlockchainTests/GeneralStateTests/VMTests/vmPerformance/loopExp.json,loopExp_d9g0v0_Shanghai
```
Comment (tothtamas28) [inline tests/slow.llvm:90]: Ending newline got removed here.

**Review by tothtamas28 — APPROVED** [review]

File: kevm-pyk/src/tests/integration/test_conformance.py (hunk 2 above)
Comment (anvacaru) [reply kevm-pyk/src/tests/integration/test_conformance.py]: This is throwing `KeyError`s for all the test files that are not in the `skipped` dict.

Comment (tothtamas28) [reply kevm-pyk/src/tests/integration/test_conformance.py]: Makes sense:

```python3
if test_name in SKIPPED_TESTS.get(gst_file, []):
    continue
```

Comment (anvacaru) [reply kevm-pyk/src/tests/integration/test_conformance.py]: Right, forgot about this for a moment. Pushing now.

**Comment by anvacaru** [comment]
test_pyk_prove[functional/lemmas-spec.k] seems to be failing with 
```
Could not verify pattern - Array [Object (fromList [("error",String "Expected a predicate but found a term"),("term",Object (fromList [("format",String "KORE"),("term",Object (fromList
```
This does not happen on master branch.

**Pull request merged**